<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ColorOpacityEditorWidget</class>
 <widget class="QWidget" name="ColorOpacityEditorWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>334</width>
    <height>774</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="mainLayout">
   <property name="spacing">
    <number>6</number>
   </property>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="1,0">
     <property name="spacing">
      <number>0</number>
     </property>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout" stretch="1,0">
       <property name="spacing">
        <number>0</number>
       </property>
       <property name="sizeConstraint">
        <enum>QLayout::SetMinimumSize</enum>
       </property>
       <item>
        <widget class="pqTransferFunctionWidget" name="OpacityEditor" native="true">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>80</height>
          </size>
         </property>
        </widget>
       </item>
       <item>
        <widget class="pqTransferFunctionWidget" name="ColorEditor" native="true">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>40</width>
           <height>40</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>40</height>
          </size>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <property name="spacing">
        <number>1</number>
       </property>
       <item>
        <widget class="QToolButton" name="ResetRangeToData">
         <property name="toolTip">
          <string>Rescale to data range</string>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="icon">
          <iconset resource="../../../Components/Resources/pqComponents.qrc">
           <normaloff>:/pqWidgets/Icons/pqResetRange24.png</normaloff>:/pqWidgets/Icons/pqResetRange24.png</iconset>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="ResetRangeToCustom">
         <property name="toolTip">
          <string>Rescale to custom range</string>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="icon">
          <iconset resource="../../../Components/Resources/pqComponents.qrc">
           <normaloff>:/pqWidgets/Icons/pqResetRangeCustom24.png</normaloff>:/pqWidgets/Icons/pqResetRangeCustom24.png</iconset>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="ResetRangeToDataOverTime">
         <property name="toolTip">
          <string>Rescale to data range over all timesteps</string>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="icon">
          <iconset resource="../../../Components/Resources/pqComponents.qrc">
           <normaloff>:/pqWidgets/Icons/pqResetRangeTemporal24.png</normaloff>:/pqWidgets/Icons/pqResetRangeTemporal24.png</iconset>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="ResetRangeToVisibleData">
         <property name="toolTip">
          <string>Rescale to visible range</string>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="icon">
          <iconset resource="../../../Components/Resources/pqComponents.qrc">
           <normaloff>:/pqWidgets/Icons/pqResetToVisibleRange32.png</normaloff>:/pqWidgets/Icons/pqResetToVisibleRange32.png</iconset>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="InvertTransferFunctions">
         <property name="toolTip">
          <string>Invert the transfer functions</string>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="icon">
          <iconset resource="../../../Components/Resources/pqComponents.qrc">
           <normaloff>:/pqWidgets/Icons/pqInvert24.png</normaloff>:/pqWidgets/Icons/pqInvert24.png</iconset>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>0</width>
           <height>1</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QToolButton" name="ChoosePreset">
         <property name="toolTip">
          <string>Choose preset</string>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="icon">
          <iconset resource="../../../Components/Resources/pqComponents.qrc">
           <normaloff>:/pqWidgets/Icons/pqFavorites16.png</normaloff>:/pqWidgets/Icons/pqFavorites16.png</iconset>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="SaveAsPreset">
         <property name="toolTip">
          <string>Save to preset</string>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="icon">
          <iconset resource="../../../Components/Resources/pqComponents.qrc">
           <normaloff>:/pqWidgets/Icons/pqSave16.png</normaloff>:/pqWidgets/Icons/pqSave16.png</iconset>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="AdvancedButton">
         <property name="toolTip">
          <string>Manually edit transfer functions</string>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="icon">
          <iconset resource="../../../Components/Resources/pqComponents.qrc">
           <normaloff>:/pqWidgets/Icons/pqAdvanced26.png</normaloff>:/pqWidgets/Icons/pqAdvanced26.png</iconset>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout" stretch="0,1">
     <item>
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Data:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="pqDoubleLineEdit" name="CurrentDataValue">
       <property name="sizePolicy">
        <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QCheckBox" name="UseLogScale">
     <property name="toolTip">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;When checked, the mapping from data to colors is done using a log-scale. Note that this does not affect the mapping of data to opacity.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
     <property name="text">
      <string>Use log scale when mapping data to colors</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QCheckBox" name="EnableOpacityMapping">
     <property name="text">
      <string>Enable opacity mapping for surfaces</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QCheckBox" name="UseLogScaleOpacity">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="text">
      <string>Use log scale when mapping data to opacity</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="ColorLabel">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Color transfer function values</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="pqExpandableTableView" name="ColorTable">
     <property name="editTriggers">
      <set>QAbstractItemView::DoubleClicked</set>
     </property>
     <property name="alternatingRowColors">
      <bool>true</bool>
     </property>
     <property name="selectionBehavior">
      <enum>QAbstractItemView::SelectRows</enum>
     </property>
     <property name="maximumRowCountBeforeScrolling" stdset="0">
      <number>10</number>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="OpacityLabel">
     <property name="text">
      <string>Opacity transfer function values</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="pqExpandableTableView" name="OpacityTable">
     <property name="editTriggers">
      <set>QAbstractItemView::DoubleClicked</set>
     </property>
     <property name="alternatingRowColors">
      <bool>true</bool>
     </property>
     <property name="selectionBehavior">
      <enum>QAbstractItemView::SelectRows</enum>
     </property>
     <property name="maximumRowCountBeforeScrolling" stdset="0">
      <number>10</number>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>pqExpandableTableView</class>
   <extends>QTableView</extends>
   <header>pqExpandableTableView.h</header>
  </customwidget>
  <customwidget>
   <class>pqTransferFunctionWidget</class>
   <extends>QWidget</extends>
   <header>pqTransferFunctionWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>pqDoubleLineEdit</class>
   <extends>QLineEdit</extends>
   <header>pqDoubleLineEdit.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../../Components/Resources/pqComponents.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>EnableOpacityMapping</sender>
   <signal>toggled(bool)</signal>
   <receiver>UseLogScaleOpacity</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>166</x>
     <y>279</y>
    </hint>
    <hint type="destinationlabel">
     <x>166</x>
     <y>305</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
